src/compiler/checker.ts

```diff
@@ -2989,12 +2983,9 @@ namespace ts {
         }
 
         function typePredicateToString(typePredicate: TypePredicate, enclosingDeclaration?: Declaration, flags?: TypeFormatFlags): string {
-            const writer = getSingleLineStringWriter();

```

Comment:
That would only happen if one of these 3 functions (`symbolToString`, `signatureToString`, `typePredicateToString`) called another one. I added an assert to prevent us from doing that.